lib/Core/Site/DomainObjectMapper.php

```diff
@@ -22,6 +23,8 @@
  */
 final class DomainObjectMapper
 {
+    use ContainerAwareTrait;
```

Comment:
I can't inject the Site directly because of circular dependency mess, it would still be called upon instantiation. If you have a suggestion how to get rid of that without exploding everything in the process, I'm all ears :)

I can inject the container through the constructor if you prefer that, it should also work.